src/index.ts

```diff
@@ -27,10 +40,27 @@ app.put("/post/:id/views", async (req, res) => {
 
 app.put("/publish/:id", async (req, res) => {
   const { id } = req.params;
-
+  let newValue;
   try {
     // toggle the `published` field on the specified post
-    const updatedPost = await prisma;
+    const post = await prisma.post.findFirst({
+      where:{
+        id: Number(id)
+      }
+    })
+    if (post?.published === false) {
+      newValue = true;
+    } else {
+      newValue = false;
+    }
+    const updatedPost = await prisma.post.update({
+      where: {
+        id: Number(id)
+      },
+      data: {
+        published: newValue,
```

Comment:
a better way to toggle would be to run `!post.published` instead of using an `if` statement. Please change it.